app/channel.go

```diff
@@ -1720,12 +1720,23 @@ func (a *App) RemoveUserFromChannel(userIdToRemove string, removerUserId string,
 	return nil
 }
 
-func (a *App) GetNumberOfChannelsOnTeam(teamId string) (int, *model.AppError) {
+func (a *App) GetNumberOfChannelsOnTeam(teamId string, includeDeleted bool) (int, *model.AppError) {
 	// Get total number of channels on current team
 	list, err := a.Srv.Store.Channel().GetTeamChannels(teamId)
```

Comment:
Maybe we should add the `includeDeleted` to the store directly to make the query more lightweight for avoid problem in crowded instances without restriction in the number of channels (or a huge number of channels limit).

Reply:
Good point but I'll leave it as is for now and we can revisit in the future if needed